### PR TITLE
FIXES #3072

### LINKED ISSUE #3072: AADGroup: MailNickname, SecurityEnabled and MailEnabled parameters should be 'required'
#### Details of the scenario you tried and the problem that is occurring

When creating a new AADGroup, the following parameters are all required by Graph, but are not required by the Microsoft365DSC module:

- MailNickname
- SecurityEnabled
- MailEnabled

See https://learn.microsoft.com/en-us/graph/api/group-post-groups?view=graph-rest-1.0&tabs=http#request-body, verified with testing

#### Verbose logs showing the problem

(For mailEnabled - the others are also true though)

VERBOSE: [COMPUTERNAME]:                            [[AADGroup]GroupName] Performing the operation "New-MgGroup_CreateExpanded" on target "Call
 remote 'GroupsGroupCreateGroup' operation".
A value is required for property 'mailEnabled' of resource 'Group'.
    + CategoryInfo          : InvalidOperation: ({ body = Micros...oftGraphGroup }:) [], CimException
    + FullyQualifiedErrorId : Request_BadRequest,Microsoft.Graph.PowerShell.Cmdlets.NewMgGroup_CreateExpanded
    + PSComputerName        : localhost

#### Suggested solution to the issue

The module should be updated so that compilation fails if these are not supplied (make them Required)

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
Configuration Example
{
    param
    (
        [Parameter(Mandatory = $true)]
        [PSCredential]
        $credsGlobalAdmin
    )

    Import-DscResource -ModuleName Microsoft365DSC

    node localhost
    {
        AADGroup 'TestGroup'
        {
            DisplayName                   = 'TestGroup'
            MailNickname                  = 'TestGroup'
            SecurityEnabled               = $true
            Ensure                        = "Present"
            Credential                    = $credsGlobalAdmin
        }
    }
}
```

#### The operating system the target node is running
OsName               : Microsoft Windows 11 Enterprise
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 22621.1.amd64fre.ni_release.220506-1250
OsLanguage           : en-GB
OsMuiLanguages       : {en-GB, en-US}

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.23.308.1

### PR DESCRIPTION
**Pull request by NikCharlebois**
#### Pull Request (PR) description
* AADGroup
  * Changed the SecurityEnabled and MailEnabled parameters to become mandatory.

#### This Pull Request (PR) fixes the following issues
* FIXES #3072

### CODE REVIEW

**Comment by ykuijs** [comment]
The issue also mentions the MailNickname property. Is that property not required?

**Comment by NikCharlebois** [comment]
It is already marked as mandatory in the .psm1 and as a Key property in the .schema.mof. Anything else we're missing?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,10 @@
 # Change log for Microsoft365DSC
 
-# UNRELEASED
+# 1.23.405.1
```
Comment (ykuijs) [inline CHANGELOG.md:3]: Don't we want to change this value until we officially release?

**Review by ykuijs — COMMENTED** [review]
One small comment

**Review by ykuijs — APPROVED** [review]
LGTM

**Pull request merged**